pkg/executor/executortype/poolmgr/gpm.go

```diff
@@ -159,18 +157,24 @@ func MakeGenericPoolManager(ctx context.Context,
 		poolPodC:                   poolPodC,
 		podSpecPatch:               podSpecPatch,
 		objectReaperIntervalSecond: time.Duration(executorUtils.GetObjectReaperInterval(logger, fv1.ExecutorTypePoolmgr, 5)) * time.Second,
+		podLister:                  make(map[string]corelisters.PodLister),
+		podListerSynced:            make(map[string]k8sCache.InformerSynced),
+	}
+	for ns, informerFactory := range gpmInformerFactory {
+		gpm.podLister[ns] = informerFactory.Core().V1().Pods().Lister()
+		gpm.podListerSynced[ns] = informerFactory.Core().V1().Pods().Informer().HasSynced
 	}
-	gpm.podLister = podInformer.Lister()
-	gpm.podListerSynced = podInformer.Informer().HasSynced
 
 	gpm.logger.Debug("inside MakeGenericPoolManager")
 
 	return gpm, nil
 }
 
 func (gpm *GenericPoolManager) Run(ctx context.Context) {
-	if ok := k8sCache.WaitForCacheSync(ctx.Done(), gpm.podListerSynced); !ok {
-		gpm.logger.Fatal("failed to wait for caches to sync")
+	for _, podListerSynced := range gpm.podListerSynced {
```

Comment:
Please form a list for all synced variables and then call `WaitForCacheSync` instead of calling in for loop. Similar to other places.